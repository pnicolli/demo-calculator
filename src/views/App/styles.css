.app {
    text-align: center;
}

.app-content {
    width: 300px;
    margin: 20px auto;
    padding: 15px 0;
    border-radius: 3px;
    background-color: #F5F5F5;
    border: 1px solid #EEEEEE;
}

.app-header {
    font-family: "Open Sans", sans-serif;
}
