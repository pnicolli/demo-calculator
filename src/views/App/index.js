import React from 'react';
import './styles.css';
import Display from '../../views/Display';
import Controls from '../../views/Controls';

const App = ({
  currentValue,
  controls,
}) => (
  <div className="app">
    <div className="app-header">
      <h1>Demo - Calculator</h1>
    </div>
    <div className="app-content">
      <Display value={currentValue} />
      <Controls controls={controls} />
    </div>
  </div>
);

App.propTypes = {
  currentValue: Display.propTypes.value,
  controls: Controls.propTypes.controls,
};

export default App;
