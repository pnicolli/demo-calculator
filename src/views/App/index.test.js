import React from 'react';
import { shallow } from 'enzyme';
import App from './index';

const baseProps = {
  currentValue: '42',
  controls: [{ text: '+', onClick: () => {} }],
};

it('matches previous snapshot', () => {
  const tree = shallow(<App {...baseProps} />);
  expect(tree).toMatchSnapshot();
});
