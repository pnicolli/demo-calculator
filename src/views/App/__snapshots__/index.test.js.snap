// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches previous snapshot 1`] = `
<div
  className="app"
>
  <div
    className="app-header"
  >
    <h1>
      Demo - Calculator
    </h1>
  </div>
  <div
    className="app-content"
  >
    <Display
      value="42"
    />
    <Controls
      controls={
        Array [
          Object {
            "onClick": [Function],
            "text": "+",
          },
        ]
      }
    />
  </div>
</div>
`;
